python_modules/dagster/dagster/core/storage/alembic/versions/030_add_columns_action_type_and_selector_id_.py

```diff
@@ -0,0 +1,43 @@
+"""add columns action_type and selector_id to bulk_actions
+
+Revision ID: 6860f830e40c
+Revises: 721d858e1dda
+Create Date: 2022-05-20 15:00:01.260860
+
+"""
+import sqlalchemy as db
+from alembic import op
+
+from dagster.core.storage.migration.utils import has_column, has_index, has_table
+
+# revision identifiers, used by Alembic.
+revision = "6860f830e40c"
+down_revision = "721d858e1dda"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    if not has_table("bulk_actions"):
```

Comment:
@prha I'm confused why this is required for the test to pass (without it I get `sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) no such table: bulk_actions`). Why does it get called when the table doesn't exist?

Reply:
This has to do with the fact that we support both maintaining the run/event_log/schedule storages in the same logical db as well as different logical dbs.  Because the alembic table (used for migration bookkeeping) is the same for each logical db, it has to support both cases.  This means that every migration script has to be applicable to every dagster storage.

In the split case, this bulk actions migration might be applied to a schedule storage, which doesn't have a bulk actions table, and so will fail.